### DISCUSSION TITLE
Xstate

### DISCUSSION THREAD

**Discussion by crystalthoughts** [ai/nanostores#252] [General]
Hi, I think this project is a great approach to what the other frameworks are converging on, and I like that its a separate module (allows for changing between component frameworks much more simply).

I was wondering if anyone tried using this with XState, and if you they any recommended workflows. I'll try it myself too.

**Comment by a-y-u-s-h** [comment]
Same thoughts.

**Comment by nicko-winner** [comment]
XState and nanostores are different use cases they are solving for. I would use Xstate if you are having issues managing state complexity, however XState it is a much more complex tool for managing the complexity of modeling and visualizing state changes. The xState state charts can help team get on same page, and understand the business logic of your app better. However  your team will not thank you for the extra complexity xState introduces, and the new mental model it introduces unless they are running into issues managing state complexity. Once past the learning curve XState can do an amazing job helping developers understand how states function, and what the possible valid states are from each state transition. 

I have not used both nano stores and xState together, but if I did, I would model my complex state in xState and manage my simple state in nano stores, and explore the idea of sharing machines in the global space via nano-stores Atoms, rather than trying to share my state machines through React context. Jotai which is similar to nano stores in many ways, just not cross-framework, developed some of these ideas further and created an extension for managing xState machines in Jotai. https://jotai.org/docs/extensions/xstate
